docs/api/browser-window.md

```diff
@@ -116,302 +84,46 @@ The [Page Visibility API][page-visibility-api] works as follows:
 It is recommended that you pause expensive operations when the visibility
 state is `hidden` in order to minimize power consumption.
 
-## Platform notices
+## `beforeunload` event
 
-* On macOS modal windows will be displayed as sheets attached to the parent window.
-* On macOS the child windows will keep the relative position to parent window
-  when parent window moves, while on Windows and Linux child windows will not
-  move.
-* On Linux the type of modal windows will be changed to `dialog`.
-* On Linux many desktop environments do not support hiding a modal window.
+For instances of `BrowserWindow`, the `close` event is emitted before the
+`beforeunload` and `unload` event of the DOM.
 
-## Class: BrowserWindow
+Usually you would want to use the `beforeunload` handler to decide whether the
+window should be closed, which will also be called when the window is reloaded.
+In Electron, returning any value other than `undefined` would cancel the close.
+For example:
 
-> Create and control browser windows.
+```javascript
+window.onbeforeunload = (e) => {
+  console.log('I do not want to be closed')
 
-Process: [Main](../glossary.md#main-process)
+  // Unlike usual browsers that a message box will be prompted to users,
+  // returning a non-void value will silently cancel the close. It is
+  // recommended to use the dialog API to let the user confirm closing the
+  // application.
+  e.returnValue = false // equivalent to `return false` but not recommended
+}
+```
+_**Note**: There is a subtle difference between the behaviors of `window.onbeforeunload = handler` and `window.addEventListener('beforeunload', handler)`. It is recommended to always set the `event.returnValue` explicitly, instead of only returning a value, as the former works more consistently within Electron._
+
+## Class: BrowserWindow extends `TopLevelWindow`
```

Comment:
It's sort of strange that TopLevelWindow gets backticks here but BrowserWindow doesn't

Reply:
This follows the style of structures, which use backticks to reference base objects (for example `MouseInputEvent`), I don't know why backticks was used there. /cc @MarshallOfSound